state-chain/custom-rpc/src/lib.rs

```diff
@@ -1171,6 +1244,11 @@ where
 		};
 
 		let mut previous = initial.clone();
+		let initial_update = SubscriptionUpdate {
```

Comment:
I believe you can factor this out into a closure. I.e.

```
|info| async {
    f(&self.client.runtime_api(), info.best_hash).map(|update| {
        SubscriptionUpdate {
			block_hash: info.hash,
			block_number: info.number(),
			data: update,
		})
    })
}
```

Reply:
I think the initial value is obtained from client info, and subsequent info are obtained from incoming Block, these are not the same type so it would be very difficult to refactor them into shared code